source/opt/const_folding_rules.cpp

```diff
@@ -350,14 +350,63 @@ ConstantFoldingRule FoldFPBinaryOp(BinaryScalarFoldingRule scalar_rule) {
   return [scalar_rule](IRContext* context, Instruction* inst,
                        const std::vector<const analysis::Constant*>& constants)
              -> const analysis::Constant* {
+    analysis::ConstantManager* const_mgr = context->get_constant_mgr();
+    analysis::TypeManager* type_mgr = context->get_type_mgr();
+    const analysis::Type* result_type = type_mgr->GetType(inst->type_id());
+    const analysis::Vector* vector_type = result_type->AsVector();
+
     if (!inst->IsFloatingPointFoldingAllowed()) {
       return nullptr;
     }
-    if (inst->opcode() == SpvOpExtInst) {
-      return FoldFPBinaryOp(scalar_rule, inst->type_id(),
-                            {constants[1], constants[2]}, context);
+
+    // Workraround a crash caused by DXC using null constants
+    if (constants[0] == nullptr || constants[1] == nullptr ||
+        constants[0]->AsNullConstant() || constants[1]->AsNullConstant()) {
+      return nullptr;
+    }
```

Comment:
I would like to see a test where this crash would happen.  I believe this could be a bug somewhere else in the folding code.  We should be able to handle the null constant.  It is just a 0.